File: Kanban.Api/DTOs/BoardDto.cs
namespace Kanban.Api.DTOs
{
    public class BoardDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string OwnerId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Nested columns
        public List<ColumnDto> Columns { get; set; } = new();
    }
}

File: kanban.client/src/dtos/BoardDto.ts
// src/dtos/BoardDto.ts
import type { ColumnDto } from './ColumnDto';

export interface BoardDto {
    id: number;
    name: string;
    description: string;
    ownerId: string;
    createdAt: string;
    updatedAt: string;
    columns: ColumnDto[];
}
